### PR TITLE
VMS-19: Simplify the creating of the shift

### PR DESCRIPTION
**Pull request by ahmedsabie**
I made changes to the following files:
shift/templates/shift/create.html
shift/templates/shift/edit.html
shift/views.py

I made the job names and dates disabled inputs.
create

edit

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
@ahmedsabie Comments [here](https://github.com/systers/vms/pull/176#issuecomment-165319539) are also applicable to this.

File: vms/shift/services.py [hunk 1]
```diff
@@ -326,15 +326,15 @@ def get_shifts_with_open_slots(j_id):
...
     shift_list = []
-
+    current_date = date.today()
     for shift in shift_list_by_date:
         slots_remaining = get_shift_slots_remaining(shift.id)
-        if slots_remaining > 0 and not is_signed_up(v_id, shift.id):
+        if slots_remaining > 0 and not is_signed_up(v_id, shift.id) and shift.date >= current_date:
             shift_map = {}
             shift_map["id"] = shift.id
             shift_map["date"] = shift.date
```
Comment (tapasweni-pathak) [inline vms/shift/services.py]: Again this should go in separate PR, fixing #130.

File: vms/event/views.py [hunk 2]
```diff
@@ -115,6 +116,7 @@ def list_sign_up(request, volunteer_id):
                 )
     else:
         event_list = get_events_ordered_by_name()
+        event_list = remove_empty_events(event_list, volunteer_id)
         return render(
             request,
             'event/list_sign_up.html',
```
Comment (tapasweni-pathak) [inline vms/event/views.py]: Changes in /event/services.py, /event/views.py should not come in this PR. This work is to fix #175. So should be present only in PR #176.

File: vms/job/services.py [hunk 3]
```diff
@@ -52,3 +52,13 @@ def get_jobs_by_event_id(e_id):
...
+
+
+def remove_empty_jobs(job_list, volunteer_id):
+    """ Removes all jobs from a job list without shifts """
+    new_job_list = []
+    for job in job_list:
+        shift_list = get_shifts_with_open_slots_for_volunteer(job.id, volunteer_id)
+        if shift_list:
+            new_job_list.append(job)
+    return new_job_list
```
Comment (tapasweni-pathak) [inline vms/job/services.py]: This change should also come only in PR #176.

File: vms/job/views.py [hunk 4]
```diff
@@ -136,6 +136,7 @@ def list_sign_up(request, event_id, volunteer_id):
         event = get_event_by_id(event_id)
         if event:
             job_list = get_jobs_by_event_id(event_id)
+            job_list = remove_empty_jobs(job_list, volunteer_id)
             return render(request, 'job/list_sign_up.html', {'event': event, 'job_list': job_list, 'volunteer_id': volunteer_id})
         else:
             raise Http404
```
Comment (tapasweni-pathak) [inline vms/job/views.py]: This change should also come only in PR #176.

**Comment by tapasweni-pathak** [comment]
@ahmedsabie I see all your code changes are coming in the PR's you make after a particular change. 

You should have changes related to one issue in one PR. So, if you are submitting a lot of changes (which is awesome! :smile: ), you can create branch per issue, and then make a PR on systers:develop branch. 

Let us know if you face any issues in fixing this.

**Comment by ahmedsabie** [comment]
@tapasweni-pathak ah ok I didn't initially know that Git submits all the changes, I thought I could choose which one to send(but I guess that could cause a lot of errors). I'll remember to only include related changes. Also regarding the other commits, what exactly am I supposed to do? they both indicate changing "git clone project_url_here" to "git clone https://github.com/systers/vms"

**Comment by tapasweni-pathak** [comment]
@ahmedsabie You can rebase and then delete those commits and then can create a pull request.

Reading more about rebase, stash, delete commit will help. 

Let us know if you face any issues.
